client/state/google-apps-users/selectors.js

```diff
@@ -0,0 +1,19 @@
+function getGoogleAppsUsersState( state ) {
+	return state.googleAppsUsers;
+}
+
+export function getByDomain( state, domain ) {
```

Comment:
This is optional, but we can make this a memoized selector: 
```javascript
const getByDomain = createSelector(
    ( state, domain ) => getGoogleAppsUsersState( state ).items.filter( item => item.domain === domain ),
    ( state ) => [ getGoogleAppsUsersState( state ) ]
);
```

https://github.com/Automattic/wp-calypso/tree/master/client/lib/create-selector#create-selector